File: opensource/test/src/main/java/com/sos/scheduler/engine/test/EnvironmentFilesTest.java
package com.sos.scheduler.engine.test;

import com.sos.scheduler.engine.kernel.util.ResourcePath;
import org.junit.Test;
import org.springframework.core.io.Resource;
import org.springframework.core.io.support.PathMatchingResourcePatternResolver;

import java.io.File;
import java.io.IOException;

import static com.google.common.io.Files.createTempDir;
import static com.sos.scheduler.engine.common.system.Files.removeFile;
import static com.sos.scheduler.engine.kernel.util.Classes.springPattern;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.*;

public final class EnvironmentFilesTest {
    private static final String[] expectedNames = {"scheduler.xml", "factory.ini", "sos.ini"};

    @Test public void test() {
        File dir = createTempDir();
        try {
            assertThat(dir.list(), emptyArray());
            EnvironmentFiles.copy(new ResourcePath(EnvironmentFilesTest.class.getPackage(), "config"), dir);
            assertThat(dir.list(), arrayContainingInAnyOrder(expectedNames));
        } finally {
            for (String name: expectedNames) removeFile(new File(dir, name));
            removeFile(dir);
        }
    }

    @Test public void testSpringGetResources() throws IOException {
        PathMatchingResourcePatternResolver r = new PathMatchingResourcePatternResolver();
        Resource[] result = r.getResources(springPattern(EnvironmentFilesTest.class.getPackage(), "config/scheduler.xml"));
        assertThat(result, arrayWithSize(1));
    }
}

File: engine-test/src/test/scala/com/sos/scheduler/engine/test/EnvironmentFilesTest.scala
package com.sos.scheduler.engine.test

import com.google.common.io.Files.createTempDir
import com.sos.scheduler.engine.common.system.Files.removeFile
import com.sos.scheduler.engine.common.utils.JavaResource
import com.sos.scheduler.engine.test.EnvironmentFilesTest._
import java.io.File
import org.junit.runner.RunWith
import org.scalatest.FreeSpec
import org.scalatest.Matchers._
import org.scalatest.junit.JUnitRunner
import org.springframework.core.io.support.PathMatchingResourcePatternResolver

@RunWith(classOf[JUnitRunner])
final class EnvironmentFilesTest extends FreeSpec {

  "copy" in {
    val dir = createTempDir()
    try {
      dir.list.toSet should be ('empty)
      TestEnvironmentFiles.copy(JavaResource("com/sos/scheduler/engine/test/config"), dir)
      dir.list.toSet should equal (ExpectedNames)
    }
    finally {
      for (name ← ExpectedNames) removeFile(new File(dir, name))
      removeFile(dir)
    }
  }

  "Spring getResources" in {
    val r = new PathMatchingResourcePatternResolver
    val result = r.getResources(s"classpath*:com/sos/scheduler/engine/test/config/scheduler.xml")
    result should have size 1
  }
}

private object EnvironmentFilesTest {
  private val ExpectedNames = Set("scheduler.xml", "factory.ini", "sos.ini")
}
